pkg/secrets/secrets.go

```diff
@@ -0,0 +1,241 @@
+package secrets
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/containers/common/pkg/secrets/filedriver"
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/containers/storage/pkg/stringid"
+	"github.com/pkg/errors"
+)
+
+// ErrInvalidPath indicates that the secrets path is invalid
+var ErrInvalidPath = errors.New("Invalid secrets path")
+
+// ErrNoSuchSecret indicates that the the secret does not exist
+var ErrNoSuchSecret = errors.New("no such secret")
+
+// ErrSecretNameInUse indicates that the secret name is already in use
+var ErrSecretNameInUse = errors.New("secret name in use")
+
+// ErrInvalidSecretName indicates that the secret name is invalid
+var ErrInvalidSecretName = errors.New("invalid secret name")
+
+// secretsFile is the name of the file that the secrets database will be stored in
+var secretsFile = "secrets.json"
+
+// SecretsManager holds information on handling secrets
+type SecretsManager struct {
+	// secretsPath is the path to the db file where secrets are stored
+	secretsDBPath string
+	// lockfile is the locker for the secrets file
+	lockfile lockfile.Locker
+	// db is an in-memory cache of the database of secrets
+	db *db
+}
+
+// Secret defines a secret
+type Secret struct {
+	// Name is the name of the secret
+	Name string `json:"name"`
+	// ID is the unique secret ID
+	ID string `json:"id"`
+	// Metadata stores other metadata on the secret
+	Metadata map[string]string `json:"metadata,omitempty"`
+	// CreatedAt is when the secret was created
+	CreatedAt time.Time
+	// Driver is the driver used to store secret data
+	Driver string
+	//driverMetadata is other metadata needed to use the driver
+	DriverOptions map[string]string
+}
+
+// SecretsDriver interfaces with the secrets data store.
+// The driver stores the actual bytes of secret data, as opposed to
+// the secret metadata.
+// Currently only the unencrypted filedriver and an in-memory driver
+// used for testing are implemented.
+type SecretsDriver interface {
+	// List lists all secret ids in the secrets data store
+	List() ([]string, error)
+	// Lookup gets the secret's data bytes
+	Lookup(id string) ([]byte, error)
+	// Store stores the secret's data bytes
+	Store(id string, data []byte) error
+	// Delete deletes a secret's data from the driver
+	Delete(id string) error
+	// DriverType returns the driver type
+	DriverType() string
```

Comment:
I think we can remove that from the API.   The secrets manager already has the knowledge.